pkg/apis/security/v1beta1/types.go

```diff
@@ -0,0 +1,191 @@
+	// +optional
+	Ports []NetworkPolicyPort `json:"ports"`
+	// Rule is matched if traffic originates from workloads selected by
+	// this field. If this field is empty, this rule matches all sources.
+	// +optional
+	From []NetworkPolicyPeer `json:"from"`
+	// Rule is matched if traffic is intended for workloads selected by
+	// this field. If this field is empty or missing, this rule matches all
+	// destinations.
+	// +optional
+	To []NetworkPolicyPeer `json:"to"`
+}
+
+// NetworkPolicyPeer describes the grouping selector of workloads.
+type NetworkPolicyPeer struct {
+	// IPBlock describes the IPAddresses/IPBlocks that is matched in to/from.
+	// IPBlock cannot be set as part of the AppliedTo field
+	// Cannot be set with any other selector.
+	// +optional
+	IPBlock *IPBlock `json:"ipBlock,omitempty"`
+	// Select Pods from AntreaNetworkPolicy's Namespace as workloads in
+	// AppliedTo/To/From fields. If set with NamespaceSelector, Pods are
+	// matched from Namespaces matched by the NamespaceSelector.
+	// Cannot be set with any other selector except NamespaceSelector.
+	// +optional
+	PodSelector *metav1.LabelSelector `json:"podSelector,omitempty"`
+	// Select all Pods from Namespaces matched by this selector, as
+	// workloads in To/From fields. If set with PodSelector,
+	// Pods are matched from Namespaces matched by the NamespaceSelector.
+	// Cannot be set with any other selector except PodSelector or
+	// ExternalEntitySelector.
+	// +optional
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+	// Select ExternalEntities from AntreaNetworkPolicy's Namespace as workloads
+	// in AppliedTo/To/From fields. If set with NamespaceSelector,
+	// ExternalEntities are matched from Namespaces matched by the
+	// NamespaceSelector.
+	// Cannot be set with any other selector except NamespaceSelector.
+	ExternalEntitySelector *metav1.LabelSelector `json:"externalEntitySelector,omitempty"`
+}
+
+// IPBlock describes a particular CIDR (Ex. "192.168.1.1/24") that is allowed
+// or denied to/from the workloads matched by a Spec.AppliedTo.
+type IPBlock struct {
+	// CIDR is a string representing the IP Block
+	// Valid examples are "192.168.1.1/24".
+	CIDR string `json:"cidr"`
+}
+
+// NetworkPolicyPort describes the port and protocol to match in a rule.
+type NetworkPolicyPort struct {
+	// The protocol (TCP, UDP, or SCTP) which traffic must match.
+	// If not specified, this field defaults to TCP.
+	// +optional
+	Protocol *v1.Protocol `json:"protocol"`
+	// The port on the given protocol. This can either be a numerical
+	// or named port on a Pod. If this field is not provided, this
+	// matches all port names and numbers.
+	// TODO: extend it to include Port Range.
+	// +optional
+	Port *intstr.IntOrString `json:"port"`
+}
+
+// RuleAction describes the action to be applied on traffic matching a rule.
+type RuleAction string
+
+const (
+	// RuleActionAllow describes that rule matching traffic must be allowed.
```

Comment:
is there an LOG that indicating packets hitting the rule should be logged?

Reply:
not in the phase 0 items.. in the proposal i have added enableLogging and REJECT action for next phase